packages/kbn-openapi-bundler/src/utils/read_document.ts

```diff
@@ -7,37 +7,48 @@
  */
 
 import fs from 'fs/promises';
-import { load } from 'js-yaml';
 import { basename, extname } from 'path';
+import { load } from 'js-yaml';
 import chalk from 'chalk';
 import { logger } from '../logger';
+import { isPlainObjectType } from './is_plain_object_type';
 
 export async function readDocument(documentPath: string): Promise<Record<string, unknown>> {
-  const extension = extname(documentPath);
-
   logger.debug(`Reading ${chalk.bold(basename(documentPath))}`);
 
+  const maybeDocument = await readFile(documentPath);
+
+  if (!isPlainObjectType(maybeDocument)) {
+    throw new Error(`File at ${chalk.bold(documentPath)} is not valid OpenAPI document`);
+  }
```

Comment:
Is this condition relevant? It doesn't look like anything other than an object can be returned from `readFile`.

Reply:
Generally JSON/YAML files may contain an array, primitive values or an empty string. These files are invalid OpenAPI documents. Even not all objects are valid OpenAPI documents but that validation is omitted here since it looks for specific keys in that object.